chapter-4/e2-reversing-an-array.js

```diff
@@ -0,0 +1,25 @@
+function reverseArray(arr) {
+  let output = [];
+  for (let i = arr.length - 1; i >= 0; i--) {
+    output.push(arr[i])
+  }
+  return output;
+}
+
+function reverseArrayInPlace(arr) {
+  let endIndex = arr.length - 1;
+  let midIndex = arr.length / 2;
```

Comment:
It's worth noting that this is is often going to produce a number like 2.5, which you can address a js array with but can cause some slightly surprising behavior when things go wrong. Taking the floor in a case like this is often a good idea.